frontend/packages/metal3-plugin/src/components/dashboard/inventory-card.tsx

```diff
@@ -0,0 +1,129 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+
+import { DashboardItemProps } from '@console/internal/components/dashboards-page/with-dashboard-resources';
+import { DashboardCard } from '@console/internal/components/dashboard/dashboard-card';
+import { DashboardCardBody } from '@console/internal/components/dashboard/dashboard-card/card-body';
+import { DashboardCardHeader } from '@console/internal/components/dashboard/dashboard-card/card-header';
+import { DashboardCardTitle } from '@console/internal/components/dashboard/dashboard-card/card-title';
+import {
+  InventoryItem,
+  SimpleInventoryItem,
+} from '@console/internal/components/dashboard/inventory-card/inventory-item';
+import { K8sResourceKind, referenceForModel } from '@console/internal/module/k8s';
+import { FirehoseResource } from '@console/internal/components/utils';
+import { MachineModel } from '@console/internal/models';
+import { getNamespace } from '@console/shared';
+
+import { getHostStorage, getHostNICs, getHostCPU, getHostMachineName } from '../../selectors';
+import { getInventoryQueries, HostQuery } from './queries';
+import { BaremetalHostModel } from '../../models';
+
+const getResources = (namespace: string, machineName: string): FirehoseResource[] => [
+  {
+    isList: false,
+    namespace,
+    name: machineName,
+    kind: referenceForModel(MachineModel),
+    prop: 'machine',
+  },
+];
+
+export const InventoryCard: React.FC<InventoryCardProps> = ({
+  obj,
+  prometheusResults,
+  watchK8sResource,
+  stopWatchK8sResource,
+  resources,
+}) => {
+  const namespace = getNamespace(obj);
+  const machineName = getHostMachineName(obj);
+  const k8sResources = getResources(namespace, machineName);
+
+  React.useEffect(() => {
+    k8sResources.forEach((r) => watchK8sResource(r));
+    return () => {
+      k8sResources.forEach((r) => stopWatchK8sResource(r));
+    };
+  }, [watchK8sResource, stopWatchK8sResource, k8sResources]);
+
+  const machineAddresses = _.get(resources.machine, 'data.status.addresses', []);
+  const machineIP = _.get(machineAddresses.find((addr) => addr.type === 'InternalIP'), 'address');
+
+  const diskCount = getHostStorage(obj).length;
+  const nicCount = getHostNICs(obj).length;
+  const cpuCount = getHostCPU(obj).count;
+
+  const queries = getInventoryQueries(machineIP);
+
+  const otherPodCount = _.get(
+    prometheusResults.getIn([queries[HostQuery.NUMBER_OF_PODS], 'result'], {}),
```

Comment:
you need to start watching these queries in `React.useEffect` by calling `watchPrometheus/stopWatchPrometheusQuery` ...see https://github.com/openshift/console/blob/master/frontend/packages/metal3-plugin/src/components/dashboard/utilization-card.tsx#L60

and then to get the result it should be

`const otherPodResult = prometheusResults.getIn([queries[HostQuery.NUMBER_OF_PODS], 'result'])`

but please keep in mind that the result can be anything, its a result of a remote call so it may be `Error` (if prometheus is down etc.). You can use this function https://github.com/openshift/console/blob/master/frontend/public/components/graphs/utils.ts#L14 to parse the result. This applies for all queries.

Reply:
:man_facepalming: I think I deleted that during clean up.